src/Fantomas/CodePrinter.fs

```diff
@@ -37,14 +37,16 @@ type ASTContext =
       /// Check whether the context is inside a SynMemberDefn.Member(memberDefn,range)
       /// This is required to correctly detect the setting SpaceBeforeMember
       IsMemberDefinition: bool
+      /// Check whether the context is inside a let binding which defines a function
+      IsInsideFunctionBinding: bool
     }
     static member Default =
         { TopLevelModuleName = ""
           IsFirstChild = false; InterfaceRange = None
           IsCStylePattern = false; IsNakedRange = false
           HasVerticalBar = false; IsUnionField = false
           IsFirstTypeParam = false; IsInsideDotGet = false
-          IsMemberDefinition = false }
+          IsMemberDefinition = false; IsInsideFunctionBinding = false }
```

Comment:
Please only add things to the ASTContext as a last resort.
In this case, you get the information from SynPat, which is available in genExprSepEqPrependType.